src/chunk.c

```diff
@@ -177,9 +177,15 @@ chunk_fill(Chunk *chunk, HeapTuple tuple, TupleDesc desc)
 {
 	chunk_formdata_fill(&chunk->fd, tuple, desc);
 
-	chunk->table_id = get_relname_relid(chunk->fd.table_name.data,
-										get_namespace_oid(chunk->fd.schema_name.data, true));
-	chunk->hypertable_relid = ts_inheritance_parent_relid(chunk->table_id);
+	if (!chunk->fd.dropped)
+	{
+		chunk->table_id = get_relname_relid(chunk->fd.table_name.data,
+											get_namespace_oid(chunk->fd.schema_name.data, true));
+		chunk->hypertable_relid = ts_inheritance_parent_relid(chunk->table_id);
+
+		Assert(OidIsValid(chunk->table_id));
```

Comment:
FWIW, this assert exists in multinode and was hit because we chunks that are marked dropped=true obviously have no real data tables. But this wasn't checked/handled in master.

Reply:
I couldn't agree more, but that's a major refactoring (which I guess should have been done when this feature was introduced). 

At this point, however, I wanted to do minimal changes to continue with our rebase of the multinode branch. More refactoring, while worthwhile, will make the rebase harder. I could do it if we agree this is important (and I think it needs to happen at some point because the handling of dropped chunk is quite broken at the moment).

Reply:
Yes, agree that refactoring can be done after code rebase.